components/segmenter/src/line.rs

```diff
@@ -251,67 +240,119 @@ impl LineSegmenter {
     ///
     /// The current behavior, which is subject to change, is to use the LSTM model when available.
     ///
-    /// See also [`Self::try_new_auto_with_options_unstable`].
+    /// See also [`Self::new_auto_with_options`].
+    ///
+    /// ✨ **Enabled with the `"data"` feature.**
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "data")]
     #[cfg(feature = "auto")]
-    pub fn try_new_auto_unstable<D>(provider: &D) -> Result<Self, SegmenterError>
-    where
-        D: DataProvider<LineBreakDataV1Marker>
-            + DataProvider<LstmForWordLineAutoV1Marker>
-            + DataProvider<GraphemeClusterBreakDataV1Marker>
-            + ?Sized,
-    {
-        Self::try_new_auto_with_options_unstable(provider, Default::default())
+    pub fn new_auto() -> Self {
```

Comment:
Question: why aren't you auto-generating the impls for the globaldata constructors in this crate?

Reply:
Auto generated constructors will go through `DataProvider::load` and be fallible. This uses singletons and is infallible.